dev/validate_bom__config.py

```diff
@@ -818,15 +819,33 @@ class MonitoringConfigurator(object):
 
   def init_argument_parser(self, parser):
     """Implements interface."""
-    pass
+    parser.add_argument(
+        '--monitoring_prometheus_gateway', default=None,
+        help='If provided, and which is "prometheus",'
+             ' configure to use the gateway server at thsi URL.')
+    parser.add_argument(
+        '--monitoring_install_which', default=None,
+        help='If provided, install monitoring with these params.')
 
   def validate_options(self, options):
     """Implements interface."""
-    pass
+    if (options.monitoring_prometheus_gateway
+        and (options.monitoring_install_which != 'prometheus')):
+      raise ValueError('gateway is only applicable to '
+                       ' --monitoring_install_which="prometheus"')
 
   def add_config(self, options, script):
     """Implements interface."""
-    pass
+    if not options.monitoring_install_which:
+      return
+
+    script.append('sudo apt-get install python-dev -y')
```

Comment:
Shouldn't be necessary, I'm doing a github release for Halyard 0.26.0 that should cover this.